Makefile

```diff
@@ -28,239 +50,251 @@ else
-kind-create: export KUBECONFIG = ${PWD}/kubeconfig
-kind-create: tmp kind
-	$(KIND) create cluster --wait 5m --image kindest/node:v1.20.0
+kind-create: export KUBECONFIG = $(PWD)/kubeconfig
+kind-create: tmp docker-build kind ## Runs a k8s kind cluster with a local registry in "localhost:5000" and ports 1080 and 1443 exposed to the host
+	$(KIND) create cluster --wait 5m --image kindest/node:v1.21.0
 
-kind-delete: ## deletes the kind cluster
-kind-delete: $(KIND)
+kind-delete: ## Deletes the kind cluster and the registry
+kind-delete: kind
 	$(KIND) delete cluster
 
-kind-deploy: ## Deploys the operator in the kind cluster for testing
-kind-deploy: export KUBECONFIG = ${PWD}/kubeconfig
-kind-deploy: manifests kustomize docker-build kind
+kind-deploy: export KUBECONFIG = $(PWD)/kubeconfig
+kind-deploy: manifests kustomize ## Deploy operator to the Kind K8s cluster
 	$(KIND) load docker-image $(IMG)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/test | kubectl apply -f -
 
-kind-refresh-operator: ## Reloads the operator image into the cluster and deletes the old Pod
 kind-refresh-operator: export KUBECONFIG = ${PWD}/kubeconfig
-kind-refresh-operator: manifests kind
+kind-refresh-operator: manifests kind ## Reloads the operator image into the K8s cluster and deletes the old Pod
 	$(KIND) load docker-image $(IMG)
 	kubectl delete pod -l control-plane=controller-manager
 
-kind-undeploy: ## Removes the operator from the kind cluster
-kind-undeploy: export KUBECONFIG = ${PWD}/kubeconfig
-kind-undeploy: manifests kustomize
+kind-undeploy: export KUBECONFIG = $(PWD)/kubeconfig
+kind-undeploy: ## Undeploy controller from the Kind K8s cluster
 	$(KUSTOMIZE) build config/test | kubectl delete -f -
 
-############################
-#### refdocs generation ####
-############################
+.PHONY: kind
+KIND = $(shell pwd)/bin/kind
+kind: ## Download kind locally if necessary
+	$(call go-get-tool,$(KIND),sigs.k8s.io/kind@v0.11.1)
+
+##@ Other
 
-refdocs: $(CRD_REFDOCS) ## Generates api reference documentation from code
+refdocs: ## Generates api reference documentation from code
+refdocs: crd-ref-docs
 	$(CRD_REFDOCS) \
 		--source-path=api \
 		--config=docs/api-reference/config.yaml \
 		--templates-dir=docs/api-reference/templates/asciidoctor \
 		--renderer=asciidoctor \
 		--output-path=docs/api-reference/reference.asciidoc
+
+$(shell pwd)/bin:
+	mkdir -p $(shell pwd)/bin
+
+.PHONY: operator-sdk
+OPERATOR_SDK_RELEASE = v1.13.1
+OPERATOR_SDK = bin/operator-sdk-$(OPERATOR_SDK_RELEASE)
+operator-sdk: ## Download operator-sdk locally if necessary.
+ifeq (,$(wildcard $(OPERATOR_SDK)))
+ifeq (,$(shell which $(OPERATOR_SDK) 2>/dev/null))
+	@{ \
+	set -e ;\
+	mkdir -p $(dir $(OPERATOR_SDK)) ;\
+	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
+	curl -sSLo $(OPERATOR_SDK) https://github.com/operator-framework/operator-sdk/releases/download/v1.13.1/operator-sdk_$${OS}_$${ARCH};\
```

Comment:
download link shouldnot harcode the version and use the envvar?

Reply:
That's the only single place where the operator-sdk version is specified, so changing it there actually changes the operator-sdk version. We can set it in a variable so it is more explicit.

Reply:
if you check some lives above, the envvar is already there, and ca be used on the download link at line 289

Reply:
oh, gotcha! 8e07249